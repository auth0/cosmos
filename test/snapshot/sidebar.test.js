import React from 'react'
import renderer from 'react-test-renderer'
import { Sidebar } from '@auth0/cosmos'
import 'jest-styled-components'

describe('Sidebar', () => {
  it('renders with automation attribute', () => {
    const tree = renderer
      .create(
        <Sidebar>
          <Sidebar.Link icon="clients" label="Clients" onClick={() => {}} />
        </Sidebar>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
