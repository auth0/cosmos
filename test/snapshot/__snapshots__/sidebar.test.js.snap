// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders with automation attribute 1`] = `
.c0 {
  float: left;
  width: 160px;
}

.c0 .c2 {
  margin-right: 8px;
  vertical-align: middle;
}

.c1 {
  display: block;
  cursor: pointer;
  color: #333;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 13px;
  line-height: 1;
  padding: calc(8px / 2) 0;
  margin-bottom: 10px;
}

.c1:hover {
  color: #FF3E00;
}

.c1:hover .c2 path {
  fill: #FF3E00;
}

.c1 .c2 {
  margin-right: 8px;
}

.c1 span {
  display: inline-block;
  vertical-align: middle;
}

.c3 {
  display: inline-block;
  line-height: 1;
}

.c4 {
  display: inline-block;
  vertical-align: middle;
  line-height: 1;
}

.c4 path {
  fill: #333;
}

<div
  className="c0"
  data-cosmos-key="sidebar"
>
  <a
    className="c1"
    data-cosmos-key="sidebar.link"
    onClick={[Function]}
  >
    <i
      className="c2 c3"
      color="default"
      name="clients"
      size={18}
    >
      <svg
        className="c4"
        color="#333"
        height={18}
        viewBox="0 0 20 20"
        width={18}
      >
        <path
          d="M17.654 15.68V4.33a2.22 2.22 0 0 0-2.22-2.22H4.084A3.309 3.309 0 0 1 6.551 1h8.882a3.33 3.33 0 0 1 3.331 3.33v8.883c0 .983-.434 1.857-1.11 2.467zM13.213 3.22a3.331 3.331 0 0 1 3.33 3.331v8.883a3.33 3.33 0 0 1-3.33 3.33H4.33A3.33 3.33 0 0 1 1 15.435V6.552A3.33 3.33 0 0 1 4.33 3.22h8.883zM2.11 15.435a2.22 2.22 0 0 0 2.22 2.22h8.883a2.22 2.22 0 0 0 2.22-2.22V8.772H2.11v6.662zm0-8.883v1.11h13.323v-1.11a2.22 2.22 0 0 0-2.22-2.22H4.33a2.22 2.22 0 0 0-2.22 2.22zm8.882 0H3.22v-1.11h7.772v1.11zm2.22 0h-1.11v-1.11h1.11v1.11z"
        />
      </svg>
    </i>
    <span>
      Clients
    </span>
  </a>
</div>
`;
